pkg/deployment/pod/utils.go

```diff
@@ -0,0 +1,68 @@
+//
+// DISCLAIMER
+//
+// Copyright 2020-2021 ArangoDB GmbH, Cologne, Germany
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Copyright holder is ArangoDB GmbH, Cologne, Germany
+//
+// Author Adam Janikowski
+//
+
+package pod
+
+import (
+	api "github.com/arangodb/kube-arangodb/pkg/apis/deployment/v1"
+	"github.com/arangodb/kube-arangodb/pkg/util/errors"
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil"
+	"github.com/arangodb/kube-arangodb/pkg/util/k8sutil/inspector/service"
+	core "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func GenerateMemberEndpoint(services service.Inspector, apiObject meta.Object, spec api.DeploymentSpec, group api.ServerGroup, member api.MemberStatus) (string, error) {
+	memberName := member.ArangoMemberName(apiObject.GetName(), group)
+	svc, ok := services.Service(memberName)
+	if !ok {
+		return "", errors.Newf("Service %s not found", memberName)
+	}
+
+	return GenerateMemberEndpointFromService(svc, apiObject, spec, group, member)
+}
+
+func GenerateMemberEndpointFromService(svc *core.Service, apiObject meta.Object, spec api.DeploymentSpec, group api.ServerGroup, member api.MemberStatus) (string, error) {
+	if group.IsArangod() {
+		switch method := spec.CommunicationMethod.Get(); method {
+		case api.DeploymentCommunicationMethodDNS, api.DeploymentCommunicationMethodIP:
+			switch method {
+			case api.DeploymentCommunicationMethodDNS:
+				return k8sutil.CreateServiceDNSNameWithDomain(svc, spec.ClusterDomain), nil
+			case api.DeploymentCommunicationMethodIP:
+				if svc.Spec.ClusterIP == "" {
+					return "", errors.Newf("ClusterIP of service %s is empty", svc.GetName())
+				}
+
+				return svc.Spec.ClusterIP, nil
+			case api.DeploymentCommunicationMethodShortDNS:
```

Comment:
it is deadcode because the `method` can be only api.DeploymentCommunicationMethodDNS or api.DeploymentCommunicationMethodIP and the below `default` is also the deadcode. I think that we should use only one `switch` and don't put `switch` into  `switch`